core/src/test/java/org/apache/felix/ipojo/EventDispatcherTest.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.felix.ipojo;
+
+import java.util.Arrays;
+import java.util.List;
+import org.apache.felix.ipojo.EventDispatcher.ListenerKey;
+import org.apache.felix.ipojo.test.MockBundle;
+import org.apache.felix.ipojo.test.MockBundleContext;
+import org.apache.felix.ipojo.test.MockServiceListener;
+import org.apache.felix.ipojo.test.MockServiceReference;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+
+public class EventDispatcherTest
+{
+    private Bundle bundle = new MockBundle(getClass().getClassLoader());
+
+    @Before
+    public void setup()
+    {
+        BundleContext bundleContext = new MockBundleContext(bundle);
+        EventDispatcher.create(bundleContext);
+    }
+
+    @After
+    public void tearDown()
+    {
+        EventDispatcher.dispose();
+    }
+
+    @Test
+    public void parsingFilters()
+    {
+        // unsupported
+        parsingFilters("");
+        parsingFilters("(foo=bar)");
+        parsingFilters("(|(objectClass=org.apache.felix.ipojo.extender.ExtensionDeclaration)(objectClass=org.apache.felix.ipojo.extender.InstanceDeclaration))");
+
+        // no filter
+        parsingFilters(null);
+        parsingFilters("");
+
+        // iPojo generated filters
+        parsingFilters("(objectClass=org.apache.felix.ipojo.extender.TypeDeclaration)",
+            new ListenerKey("org.apache.felix.ipojo.extender.TypeDeclaration", null, null, null, null, null));
+        parsingFilters("(objectclass=org.apache.felix.ipojo.extender.TypeDeclaration)",
```

Comment:
isn't it the same then just above ?

Reply:
almost, except the uppercase

object**C**lass vs object**c**lass